app/assets/javascripts/validation.js

```diff
@@ -10,69 +10,183 @@ $(document).ready(function () {
-        minlength: 3
-      },
-      email: {
-        required: true,
-        email: true
-      },
-      password1: {
-        required: true,
-        minlength: 8
-      },
-      password2: {
-        equalTo: "#password1"
-      }
-    },
-    messages: {
-      username: {
-        required: "Please enter username",
-        minlength: "Username should be minimum 3 characters long"
-      },
-      password1: {
-        required: "Please enter password",
-        minlength: "Password should be minimum 8 characters long"
-      },
-      email: {
-        required: "Please enter email",
-        email: "Invalid email address"
-      },
-      password2: {
-        required: "Please enter password",
-        minlength: "Password should be minimum 8 characters long",
-        equalTo: "Passwords doesn't match"
-      }
-    },
-    submitHandler: function(form) {
-      form.submit();
+$(document).ready(function () {
+  var signUpForm = document.querySelector('#create-form');
+
+  if (!signUpForm) return;
+
+  var validationTracker = {};
+  var submitBtn = document.querySelector("#create-form [type='submit']");
+  var usernameElement = document.querySelector("[name='user[username]']");
+  var emailElement = document.querySelector("[name='user[email]']");
+  var passwordElement = document.querySelector("[name='user[password]']");
+  var confirmPasswordElement = document.querySelector(
+    "[name='user[password_confirmation]']"
+  );
+
+  isFormValid();
+
+  // Every time user types something, it triggers corresponding event listener
+  usernameElement.addEventListener('input', validateUsername);
+  emailElement.addEventListener('input', validateEmail);
+  passwordElement.addEventListener('input', validatePassword);
+  confirmPasswordElement.addEventListener('input', validateConfirmPassword);
+
+  function validateUsername(e) {
+    var username = e.target.value;
+
+    if (username.length < 3) {
+      restoreOriginalStyle(this);
+    } else {
+      $.get('/api/srch/profiles?query=' + username, function(data) {
+        if (data.items) {
+          $.map(data.items, function(userData) {
+            if (userData.doc_title === username) {
+              updateUI(usernameElement, false, 'Username already exists');
+            } else {
+              updateUI(usernameElement, true);
+            }
+          });
+        } else {
+          updateUI(usernameElement, true);
+        }
+      });
     }
-  });
+  }
+
+  function validateEmail(e) {
+    var email = e.target.value;
+    var regexp = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    var isValid = regexp.test(email);
+
+    updateUI(emailElement, isValid, 'Invalid email');
+  }
+
+  function validatePassword(e) {
+    var password = e.target.value;
+
+    if (!isPasswordValid(password)) {
+      updateUI(this, false, 'Minimum 8 characters including 1 number');
```

Comment:
Yes, I could change this, but then it would look like this:

![Screenshot_20200103_020551](https://user-images.githubusercontent.com/35997844/71702067-afd8fb80-2dcd-11ea-8007-41c3f168dea0.png)

Reply:
Kindly make it
Atleast one number

Reply:
Got it :+1: